numpy/core/src/umath/ufunc_object.c

```diff
@@ -5893,13 +5893,45 @@ new_array_op(PyArrayObject *op_array, char *data)
     return (PyArrayObject *)r;
 }
 
+/*
+ * If operands are 1D we can use the indexed loop to do the work
+ * Returns 0 if successful, -1 otherwise
+ */
+static int
+try_trivial_at_loop(PyArrayMethodObject *ufuncimpl, NPY_ARRAYMETHOD_FLAGS flags,
+                    PyArrayMapIterObject *iter,
+                    PyArrayObject *op1_array, PyArrayObject *op2_array,
+                    PyArrayMethod_Context *context)
+{
+    if (PyArray_NDIM(op1_array) != 1) {
+        return -1;
+    }
+    /* check that nop == 3 */
```

Comment:
This comment seems wrong: it just checks that the dimension of any second array is 1.

Reply:
I changed this to `trivial_at_loop` and moved all the checks to the caller.